### PR TITLE
More typescript migration

### PR DESCRIPTION
**Pull request by jespino**
This migrates some small and simple files to typescript, 34 files left until full migration is complete :)

### CODE REVIEW

File: webapp/src/plugin.tsx [hunk 1]
```diff
@@ -15,7 +18,7 @@ import reducers from './reducers';
 import {handleConnectChange, getConnected, handleInstanceStatusChange, getSettings} from './actions';
 import Hooks from './hooks/hooks';
 
-const setupUILater = (registry, store) => async () => {
+const setupUILater = (registry: PluginRegistry, store: Store<object, Action<object>>): () => Promise<void> => async () => {
```
Comment (jespino) [inline webapp/src/plugin.tsx]: The store definition here is a bit generic, but for now I think is enough

**Comment by mickmister** [comment]
@DHaussermann stale PR waiting on QA review

**Comment by levb** [comment]
Shouldn't we have changed https://github.com/mattermost/mattermost-plugin-jira/blob/master/build/manifest/main.go#L113 with this PR? @mickmister @cpoile

**Comment by mickmister** [comment]
@levb Makes sense though I'm not familiar with what the function `applyManifest` does. I wonder why there were no issues with deploying with that there. It might be writing to that name for something else expecting a js file.

**Comment by levb** [comment]
Fixed with https://github.com/mattermost/mattermost-plugin-jira/pull/323

**Comment by mickmister** [comment]
@levb Is the change solving an issue that you have experienced? If something else relies on the js file name, it will break.

**Comment by levb** [comment]
@jespino renamed the generated manifest.js->manifest.ts so there was one in the code, it just wasn't getting updated by `make`. `make` continued to generate a `manifest.js` - I was puzzled as to why it popped up as a new file, that's why I asked.

**Pull request merged**